src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1103,27 +1091,8 @@ internal void ExecuteSubmission(BuildSubmission submission, bool allowMainThread
                         return;
                     }
 
-                    // Submit the build request.
-                    _workQueue.Post(
-                        () =>
-                        {
-                            try
-                            {
-                                IssueBuildSubmissionToScheduler(submission, allowMainThreadBuild);
-                            }
-                            catch (BuildAbortedException bae)
-                            {
-                                // We were canceled before we got issued by the work queue.
-                                var result = new BuildResult(submission.BuildRequest, bae);
-                                submission.CompleteResults(result);
-                                submission.CompleteLogging(true);
-                                CheckSubmissionCompletenessAndRemove(submission);
-                            }
-                            catch (Exception ex) when (!ExceptionHandling.IsCriticalException(ex))
-                            {
-                                HandleExecuteSubmissionException(submission, ex);
-                            }
-                        });
+                    AddBuildRequestToSubmission(submission, newConfiguration.ConfigurationId);
```

Comment:
It was in preparation for the next phase, handling cache requests (looking at all diffs makes it more clear). Cache requests do not need a build request set on their submission. Build submissions that undergo cache requests have a build request set on them only after their cache result is posted back to the BM (real build request for cache misses, proxy build requests for cache hits)

Reply:
@rainersigwald I replaced the potentially non-deterministic test with a test that hangs if the build manager queries the cache sequentially. Not sure which option is best, probably the hanging one because it won't hang as long as the code does not regress :)